core/src/main/java/com/linecorp/armeria/client/Endpoint.java

```diff
@@ -432,21 +445,36 @@ public String authority() {
 
         if (isGroup()) {
             authority = "group:" + groupName;
-        } else if (port != 0) {
-            if (hostType == HostType.IPv6_ONLY) {
-                authority = '[' + host() + "]:" + port;
-            } else {
-                authority = host() + ':' + port;
-            }
-        } else if (hostType == HostType.IPv6_ONLY) {
-            authority = '[' + host() + ']';
         } else {
-            authority = host();
+            authority = authorityWithoutPort();
+            if (port != 0) {
+                authority += ":" + port; // ':' will add 58, so use double quotes.
+            }
         }
 
         return this.authority = authority;
     }
 
+    /**
+     * Converts this endpoint into the authority part of a URI removing the port. The return value will be the
+     * same value from {@link #host()} except when the IP address of this endpoint is a valid IP V6 address.
+     * In that case, the value will be {@link #host()} surrounded by {@code []}. If {@link #hasPort()} returns
+     * {@code false}, this value will be the same as {@link #authority()}.
+     */
+    public String authorityWithoutPort() {
```

Comment:
With my proposed change, we do not need this method anymore. How about adding `Endpoint.withoutPort()` instead, so that a user can call `Endpoint.withoutPort().authority()`, which is more composable?

Reply:
Alternatively adding `host()` might make sense too, since I think it is the same pattern as the normal Java `URL`

https://docs.oracle.com/javase/7/docs/api/java/net/URL.html#getHost()

Reply:
`host()` is already there actually. :sweat_smile:

Reply:
Ah should have looked at the code too... Yeah `withoutPort()` seems cleaner then

Reply:
Let me just remove this method because it seems that I don't need this anymore.
Will add later when I need it. Thank!